test/test_model.py

```diff
@@ -175,6 +181,49 @@ def test_config(self):
             # Confirm that we cannot modify config values.
             self.model.config['foo'] = 'bar'
 
+    def test_is_leader(self):
+        self.backend = op.model.ModelBackend()
+        self.model = op.model.Model('myapp/0', ['db0', 'db1', 'db2'], self.backend)
+
+        def check_remote_units():
+            fake_script(self, 'relation-ids',
+                        """[ "$1" = db1 ] && echo '["db1:4"]' || echo '[]'""")
+
+            fake_script(self, 'relation-list',
+                        """[ "$2" = 4 ] && echo '["remoteapp1/0", "remoteapp1/1"]' || exit 2""")
+
+            # Cannot determine leadership for remote units.
+            for u in self.model.get_relation('db1').units:
+                with self.assertRaises(RuntimeError):
+                    u.is_leader
+
+        fake_script(self, 'is-leader', 'echo true')
```

Comment:
Have you double checked what the exact behavior of is-leader is?  That looks like a bad implementation.. I hope that's not how it works at least.

Reply:
Yes, for reference (`2.6.8`):

```
juju run --unit keystone/0 'is-leader --format=json'
false
```

Per https://tools.ietf.org/html/rfc7159 this is valid JSON:

```
JSON-text = ws value ws

value = false / null / true / object / array / number / string

ws = *(
        %x20 /              ; Space
        %x09 /              ; Horizontal tab
        %x0A /              ; Line feed or New line
        %x0D )              ; Carriage return
```

There is even [an example](https://tools.ietf.org/html/rfc7159#section-13) at the end of it:

```
   Here are three small JSON texts containing only values:

   "Hello world!"

   42

   true
```

Are you suggesting that old parsers might expect objects or arrays for the "top-level" value?

Reply:
Per the discussion during the call: is-leader returns true/false instead of 0/1 exit codes so nothing we can do about it here.